ctapipe/io/eventsource.py

```diff
@@ -198,3 +226,10 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         pass
+
+
+def from_config(config, *args, **kwargs):
```

Comment:
Noooo :-D Single Responsibility Principle.
No seriously. Sure it could .. but I don't think it is a good idea to do so. `event_source` has a pretty clear interface. It takes a `url` an returns an instance of a compatible based only on that `url`. (We ignore that one can provide more parameters which are just forwarded into the particular EventSources __init__, since it is transparent) 

`from_config` is a complex thing dealing with traitlets.config.LazyConfig objects. It might break for all sorts of reasons and it might need to be modified for all sorts traitlets connected reasons. Single Responsibility means not only "one purpose" but also "one reason to change" ...

Reply:
Okay, this makes sense